cardano-cli/src/Cardano/CLI/Byron/Key.hs

```diff
@@ -134,17 +122,11 @@ readPaymentVerificationKey (VerificationKeyFile fp) = do
   -- Convert error to 'CliError'
   firstExceptT (VerificationKeyDeserialisationFailed fp . T.pack . show) eVk
 
-
-serialisePoorKey :: ByronKeyFormat -> Genesis.PoorSecret
-                 -> Either ByronKeyFailure LB.ByteString
-serialisePoorKey bKeyFormat ps =
-  serialiseSigningKey bKeyFormat $ Genesis.poorSecretToKey ps
-
 -- | Generate a cryptographically random signing key,
 --   protected with a (potentially empty) passphrase.
-keygen :: Crypto.PassPhrase -> IO SigningKey
+keygen :: Crypto.PassPhrase -> IO (SigningKey ByronKey)
 keygen passphrase =
-  snd <$> Crypto.runSecureRandom (Crypto.safeKeyGen passphrase)
+  ByronSigningKey . snd <$> Crypto.runSecureRandom (Crypto.safeKeyGen passphrase)
```

Comment:
Can't we replace this with `Cardano.Api.generateSigningKey`?

Reply:
Yep this was my plan, but I got sidetracked.